### ISSUE TITLE
Adapt crate to Stacked Borrows?

### ISSUE THREAD

**Issue by avl** [golddranks/multi_mut#1]
Hi,

In your introduction you mention that it is impossible to determine if the unsafe things done in this crate are UB or not. Since the release of Stacked Borrows, there is now a formal model against which things could be verified (and also miri, to do automated checking).

Are you still maintaining this crate? If so, would you be interested in working toward making things safe according to these new rules? Would you be interested in accepting pull requests?

**Comment by thomcc** [comment]
Yeah, this is unsound stacked borrows or no. It's invalid under llvm's noalias rules, which is probably why it hasn't triggered errors in the past (these were turned off for &mut until recently).
